### PR TITLE
thorium-browser: init at 117.0.5938.157

### LINKED ISSUE #184823: Thorium Browser
**Project description**
Chromium fork with performance & privacy benefits.

**Metadata**

* homepage URL: https://thorium.rocks/
* source URL: https://github.com/Alex313031/Thorium
* license:  GPL-3.0
* platforms: Linux, Darwin

### PR DESCRIPTION
**Pull request by isabelroses**
## Description of changes

- Added [thorium browser](https://github.com/Alex313031/thorium) at version 117.0.5938.157

closes https://github.com/NixOS/nixpkgs/issues/184823

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Review by eclairevoyant — CHANGES_REQUESTED** [review]
Thanks for the contribution! Couple points below, and also please reorder the commits so that the maintainer one comes first

File: pkgs/applications/networking/browsers/thorium-browser/default.nix
Comment (FliegendeWurst) [reply pkgs/applications/networking/browsers/thorium-browser/default.nix:160]: The source is available: https://github.com/Alex313031/thorium

Looks like it uses ninja as a build tool: https://github.com/Alex313031/thorium/blob/main/build.sh

File: pkgs/top-level/all-packages.nix
Comment (eclairevoyant) [inline pkgs/top-level/all-packages.nix:13855]: this can be removed, as you're using regular callPackage with no args

**Review by eclairevoyant — CHANGES_REQUESTED** [review]

**Comment by cawilliamson** [comment]
IMHO this build should be adapted to accept the NIXOS_OZONE_WL flag in order to make it play nicely in Wayland just like Chromium does.

**Comment by A7R7** [comment]
It may be better to replace unrar with _7zz, since unrar has an unfree license. I tried replace unrar with _7zz and it can still build and run

**Comment by rgri** [comment]
Anything holding this package back that I could help with?

**Comment by rgri** [comment]
I'm getting an error trying to run `thorium-shell`:
```bash
$out/bin/..thorium-shell-wrapped-wrapped: line 3: /opt/chromium.org/thorium/thorium_shell: No such file or directory
```
which is caused by a reference to `/opt/[...]` in the relevant file. I don't know enough about the package structure to fix it right now but I'm assuming there's something missing from the `substituteInPlace` blocks in `installPhase`.

**Comment by VuiMuich** [comment]
It's probably not relevant or blocking for the merge, but I had to rebase with `master` when testing this locally, so it got correctly built with `glibc 2.38' and not `2.37`. Also getting a bunch of `dbus` errors, but I assume that's from running it straight from `nixpkgs/results/bin`..

**Comment by isabelroses** [comment]
> but I assume that's from running it straight from`nixpkgs/results/bin`..

When I personally tried running it from the desktop file, I didn't get any dbus errors. There are a few errors but I would also get those with default chromium. So I think its alright

**Comment by VuiMuich** [comment]
Yeah, doesn't seem like serious errors for me as well and I could not see anything in Thorium *not working* properly in my brief test.

**Comment by rgri** [comment]
Hm. I tested the package rebased onto `e7704a1b645c` and `thorium-shell` fails with the same error. I tested locally by adding an input to my config pointing to a local branch of nixpkgs with the latest commit on this branch rebased onto master. Unless I'm missing something, I'm pretty sure this [prevents](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#reviewing-contributions) approval.

It would be helpful if anyone could confirm or perhaps we could get some input from the dedicated @isabelroses. Eager to see this merged!

**Comment by isabelroses** [comment]
To be accurate the existence of thorium-shell is for thorium devs, the only reason I've left it here was because when the binary was missing all sorts of first paint errors etc were thrown. But I will try fix the issue.

**Comment by felixsanz** [comment]
what is blocking the merge?

**Comment by Icey-Glitch** [comment]
whats preventing merging?

**Comment by AidanJ** [comment]
There's a good reason now.

**Comment by vigress8** [comment]
Thorium has been updated

**Comment by isabelroses** [comment]
I'm really sorry to do this but I no longer have the time to properly continue this PR and fix any issues. Thanks for everyone's time helping me further this PR.

**Comment by rgri** [comment]
Hi @isabelroses . Thanks for your all your hard work. Would you mind if I picked this PR up, and built upon your work?

**Comment by isabelroses** [comment]
> Hi @isabelroses . Thanks for your all your hard work. Would you mind if I picked this PR up, and build upon your work?

Please be my guest, you may have all rights to the code as well.

**Pull request closed without merging**
